flaky test in the TestWorkerPool

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
 === RUN   TestWorkerPool/TwoWorkers
    utils_test.go:249: 
        	Error Trace:	pkg/executor/utils_test.go:249
        	Error:      	Not equal: 
        	            	expected: []int{1, 3, 2, 4}
        	            	actual  : []int{1, 2, 3, 4}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,4 +2,4 @@
        	            	  (int) 1,
        	            	+ (int) 2,
        	            	  (int) 3,
        	            	- (int) 2,
        	            	  (int) 4
        	Test:       	TestWorkerPool/TwoWorkers
=== RUN   TestWorkerPool/TolerateOnePendingTask
--- FAIL: TestWorkerPool (0.00s)
    --- PASS: TestWorkerPool/SingleWorker (0.00s)
    --- FAIL: TestWorkerPool/TwoWorkers (0.00s)
    --- PASS: TestWorkerPool/TolerateOnePendingTask (0.00s)
FAIL
```

https://tiprow.hawkingrei.com/view/gs/pingcapprow/pr-logs/pull/pingcap_tidb/59092/fast_test_tiprow/1881613739241771008#1:build-log.txt%3A10546

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/executor/utils_test.go
+++ b/pkg/executor/utils_test.go
@@ -15,9 +15,9 @@
 package executor
 
 import (
-	"runtime"
 	"sync"
 	"testing"
+	"time"
 
 	"github.com/pingcap/errors"
 	"github.com/pingcap/tidb/pkg/domain"
@@ -196,6 +196,9 @@ func TestWorkerPool(t *testing.T) {
 		list = list[:0]
 		lock.Unlock()
 	}
+	sleep := func(ms int) {
+		time.Sleep(time.Duration(ms) * time.Millisecond)
+	}
 
 	t.Run("SingleWorker", func(t *testing.T) {
 		clean()
@@ -211,11 +214,11 @@ func TestWorkerPool(t *testing.T) {
 			wg.Add(1)
 			pool.submit(func() {
 				push(3)
-				runtime.Gosched()
+				sleep(10)
 				push(4)
 				wg.Done()
 			})
-			runtime.Gosched()
+			sleep(1)
 			push(2)
 			wg.Done()
 		})
@@ -237,11 +240,11 @@ func TestWorkerPool(t *testing.T) {
 			wg.Add(1)
 			pool.submit(func() {
 				push(3)
-				runtime.Gosched()
+				sleep(10)
 				push(4)
 				wg.Done()
 			})
-			runtime.Gosched()
+			sleep(1)
 			push(2)
 			wg.Done()
 		})
@@ -263,11 +266,11 @@ func TestWorkerPool(t *testing.T) {
 			wg.Add(1)
 			pool.submit(func() {
 				push(3)
-				runtime.Gosched()
+				sleep(10)
 				push(4)
 				wg.Done()
 			})
-			runtime.Gosched()
+			sleep(1)
 			push(2)
 			wg.Done()
 		})

